File: frontend-Application/src/services/admin/courseService.js
import adminAxios from "../axios/adminAxios";

export const getAllCourses = async () => {
    return await adminAxios.get("/admin/courses");
};

export const getCourseById = async (id) => {
    return await adminAxios.get(`/admin/courses/${id}`);
};

export const createCourse = async (course) => {
    return await adminAxios.post("/admin/courses", course);
};

export const updateCourse = async (id, course) => {
    return await adminAxios.put(`/admin/courses/${id}`, course);
};

export const updateCourseStatus = async (id, status) => {
    return await adminAxios.patch(`/admin/courses/${id}/status`, { status });
};

export const deleteCourse = async (id) => {
    return await adminAxios.delete(`/admin/courses/${id}`);
};

File: oep_spring_backend/src/main/java/com/oep/service/CourseService.java
package com.oep.service;

import java.util.List;
import com.oep.entities.Courses;
import com.oep.dtos.CourseRequestDto;

public interface CourseService {
	List<Courses> getAllCourses();

	List<Courses> getCoursesByInstructor(Long instructorId);

	List<Courses> getCoursesByStudent(Long studentId);

	Courses getCourseById(Long id);

	Courses createCourse(CourseRequestDto course);

	Courses updateCourse(Long id, CourseRequestDto course);

	void updateCourseStatus(Long id, String status);

	void deleteCourse(Long id);
}
